### ISSUE TITLE
data block `aws_availability_zones` providing `names` attribute as a string

### ISSUE THREAD

**Issue by aliscott** [infracost/infracost#1856]
**Terraform code:**
```hcl
provider "aws" {
  region                      = "eu-west-1"
  skip_credentials_validation = true
  skip_requesting_account_id  = true
  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"
}

data "aws_availability_zones" "available" {}

resource "aws_vpc" "tenant_aws_vpc" {
  cidr_block       = "10.0.0.0/16"
  instance_tenancy = "default"
}

resource "aws_subnet" "public_subnet" {
  count             = length(data.aws_availability_zones.available.names)
  vpc_id            = aws_vpc.tenant_aws_vpc.id
  cidr_block        = "172.5.${10 + count.index}.0/24"
  availability_zone = data.aws_availability_zones.available.names[count.index]
}

resource "aws_vpc_endpoint" "with_dynamic_subnet" {
  vpc_id = aws_vpc.tenant_aws_vpc.id
  subnet_ids = [
    for public_subnet in aws_subnet.public_subnet : public_subnet.id
  ]
  service_name      = "com.amazonaws.us-east-1.ec2messages"
  vpc_endpoint_type = "Interface"

  security_group_ids = ["fakeid"]

  private_dns_enabled = true
}
```

**Actual result:**

This shows me the price assuming there's 22 subnets, when there's actually only 3. This looks like it's because the HCL parsing is mocking `data.aws_availability_zones.available.names` to `"availability_zone-mock"` which is a string with length 22.

```
 Name                                    Monthly Qty  Unit            Monthly Cost

 aws_vpc_endpoint.with_dynamic_subnet
 ├─ Data processed (first 1PB)         Monthly cost depends on usage: $0.01 per GB
 └─ Endpoint (Interface)                      16,060  hours                $176.66

 OVERALL TOTAL                                                             $176.66
 ```
 
 **Expected result:**
 
I think the expected result would be to match the availability zone count from AWS, but with HCL parsing this isn't possible since we don't have creds to query the cloud. So maybe a good option initially is just to default the count of arrays to 1 with HCL parsing, and in the future we can add them to the usage file or have some way of overriding them.

**Comment by ckdake** [comment]
A default of one would be nice, but as an infracost user I would be happy to either add an annotation to the file. e.g.:

```hcl
#infracost:expected-count(4)
data "aws_availability_zones" "available" {}
```

Or include something in a infracost.yml config file in / that helps this out.

**Comment by fatihtokus** [comment]
@aliscott , please assign this to me

**Comment by hugorut** [comment]
Hey @fatihtokus, It might be a good idea to outline your potential solution to this issue either here or in the `#contributors` chat.  The mocking functionality in HCL parsing is a little tricky, and there are several edge cases that we need to be aware of. So it might be useful to discuss before you set out on implementation, as I don't want you to waste time on a solution that might require tons of work or might not be accepted.

**Comment by aliscott** [comment]
I just hit a case where the code had:

```hcl
locals {
  selected_availability_zones = slice(sort(data.aws_availability_zones.available.names), 0, 2)
}

data "aws_availability_zones" "available" {}
```

For this, we actually know the length of `local.selected_availability_zones` should be 2 because of the slice function. Not sure if there's something smart we can do here.

Another option might be to mock well-known data attributes like `aws_availability_zones` with the realistic values.

**Comment by hugorut** [comment]
Hi all we've made a fix in #2504 to support static availability zones being returned from `aws_availability_zones` and `google_compute_zones`. This should solve the mocking bug outlined in this issue. The funcitonality will be part of the next release, and we'll let you know when it goes live here.
